### PR TITLE
feat(Text)!: update truncation in Text component

### PR DESCRIPTION
**Pull request by aresnik11**
### Overview

Adds `truncateLines` prop to `Text` from the `TruncateText` component in portal app. This should be used with the `truncate` prop to denote which form of truncation you would like (ellipsis or fade) and the types are set up so that you cannot use one prop without the other. Also marks `Truncate` as deprecated in favor of using this new truncation (I can remove this if people don't agree but it's confusing that we have 2 different forms of truncation).

Portal app PR: https://github.com/codecademy-engineering/portal-app/pull/1208
Monolith PR: https://github.com/codecademy-engineering/Codecademy/pull/28786

### PR Checklist

- [ ] Related to designs:
- [x] Related to JIRA ticket: [REACH-1205](https://codecademy.atlassian.net/browse/REACH-1205)
- [x] I have run this code to verify it works
- [ ] This PR includes unit tests for the code change

### CODE REVIEW

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.37-alpha.56353d.0<br/>@codecademy/gamut-labs@28.5.2-alpha.56353d.0<br/>@codecademy/gamut@42.2.2-alpha.56353d.0<br/>@codecademy/monaco@0.5.28-alpha.56353d.0<br/>@codecademy/styleguide@52.3.2-alpha.56353d.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61bce556ccad560c91ed4d49--gamut-preview.netlify.app>https://61bce556ccad560c91ed4d49--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61bce556ccad560c91ed4d49>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.37-alpha.8bf6b3.0<br/>@codecademy/gamut-labs@28.5.2-alpha.8bf6b3.0<br/>@codecademy/gamut@42.2.2-alpha.8bf6b3.0<br/>@codecademy/monaco@0.5.28-alpha.8bf6b3.0<br/>@codecademy/styleguide@52.3.2-alpha.8bf6b3.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61bce8f72671bc138a710736--gamut-preview.netlify.app>https://61bce8f72671bc138a710736--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61bce8f72671bc138a710736>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.37-alpha.8b96e7.0<br/>@codecademy/gamut-labs@28.5.2-alpha.8b96e7.0<br/>@codecademy/gamut@42.2.2-alpha.8b96e7.0<br/>@codecademy/monaco@0.5.28-alpha.8b96e7.0<br/>@codecademy/styleguide@52.3.2-alpha.8b96e7.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61bd14b6d4f9803aea8174af--gamut-preview.netlify.app>https://61bd14b6d4f9803aea8174af--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61bd14b6d4f9803aea8174af>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.a4cf8b.0<br/>@codecademy/gamut-kit@0.6.38-alpha.a4cf8b.0<br/>@codecademy/gamut-labs@28.6.1-alpha.a4cf8b.0<br/>@codecademy/gamut@42.3.0-alpha.a4cf8b.0<br/>@codecademy/monaco@0.5.28-alpha.a4cf8b.0<br/>@codecademy/styleguide@52.4.1-alpha.a4cf8b.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d4b2f595fdea1b1a7120a4--gamut-preview.netlify.app>https://61d4b2f595fdea1b1a7120a4--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d4b2f595fdea1b1a7120a4>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.1939be.0<br/>@codecademy/gamut-kit@0.6.38-alpha.1939be.0<br/>@codecademy/gamut-labs@28.6.1-alpha.1939be.0<br/>@codecademy/gamut@42.3.0-alpha.1939be.0<br/>@codecademy/monaco@0.5.28-alpha.1939be.0<br/>@codecademy/styleguide@52.4.1-alpha.1939be.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d4bb0b6392d82eef984bc0--gamut-preview.netlify.app>https://61d4bb0b6392d82eef984bc0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d4bb0b6392d82eef984bc0>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-icons@6.1.0-alpha.bb4829.0<br/>@codecademy/gamut-kit@0.6.38-alpha.bb4829.0<br/>@codecademy/gamut-labs@28.6.1-alpha.bb4829.0<br/>@codecademy/gamut@42.3.0-alpha.bb4829.0<br/>@codecademy/monaco@0.5.28-alpha.bb4829.0<br/>@codecademy/styleguide@52.4.1-alpha.bb4829.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d4dc8b256a2f500dbf49e2--gamut-preview.netlify.app>https://61d4dc8b256a2f500dbf49e2--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d4dc8b256a2f500dbf49e2>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.39-alpha.7b1394.0<br/>@codecademy/gamut-labs@28.6.2-alpha.7b1394.0<br/>@codecademy/gamut@42.3.1-alpha.7b1394.0<br/>@codecademy/monaco@0.5.29-alpha.7b1394.0<br/>@codecademy/styleguide@52.5.1-alpha.7b1394.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61d8aba2ed8d7837f96e4bc0--gamut-preview.netlify.app>https://61d8aba2ed8d7837f96e4bc0--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61d8aba2ed8d7837f96e4bc0>Deploy Logs</a></p>

File: packages/gamut/src/Typography/Text.tsx [hunk 1]
```diff
@@ -72,20 +89,33 @@ const textProps = variance.compose(
...
+  truncateLinesProps
 );
 
-export interface TextProps
+export interface BaseTextProps
   extends StyleProps<typeof textProps>,
     StyleProps<typeof textStates>,
-    StyleProps<typeof truncateVariants>,
     StyleProps<typeof elementVariants>,
+    StyleProps<typeof truncateVariants>,
```
Comment (aionate0812) [inline packages/gamut/src/Typography/Text.tsx:100]: Curious why we changed the order here and on line 118?

**Review by aionate0812 — APPROVED** [review]
Changes look good to me - agree on marking the `Truncate` component as depricated

File: packages/gamut/src/Typography/Text.tsx (hunk 1 above)
Comment (aresnik11) [reply packages/gamut/src/Typography/Text.tsx:100]: no reason, must have been unintentional but the order shouldn't matter

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.40-alpha.fbc70a.0<br/>@codecademy/gamut-labs@28.6.3-alpha.fbc70a.0<br/>@codecademy/gamut@42.3.1-alpha.fbc70a.0<br/>@codecademy/monaco@0.5.29-alpha.fbc70a.0<br/>@codecademy/styleguide@52.5.2-alpha.fbc70a.0<br/></pre>

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61ddf43892e31f2ed9997c04--gamut-preview.netlify.app>https://61ddf43892e31f2ed9997c04--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61ddf43892e31f2ed9997c04>Deploy Logs</a></p>

**Review by christian-dinh — APPROVED** [review]
it's been mentioned on other PRs but defaulting `truncateLines` to 1 would be nicer imo! still good tho

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.40-alpha.74f0f8.0<br/>@codecademy/gamut-labs@28.6.3-alpha.74f0f8.0<br/>@codecademy/gamut@42.3.1-alpha.74f0f8.0<br/>@codecademy/monaco@0.5.29-alpha.74f0f8.0<br/>@codecademy/styleguide@52.5.2-alpha.74f0f8.0<br/></pre>

**Review by dreamwasp — APPROVED** [review]
i love it

**Comment by codecademydev** [comment]
<p>🚀 Styleguide deploy preview ready!</p><p><a href=https://61df420ffdbe314fad11961d--gamut-preview.netlify.app>https://61df420ffdbe314fad11961d--gamut-preview.netlify.app</a></p><p><a href=https://app.netlify.com/sites/gamut-preview/deploys/61df420ffdbe314fad11961d>Deploy Logs</a></p>

**Comment by codecademydev** [comment]
<h3>📬Published Alpha Packages:</h3><pre>@codecademy/gamut-kit@0.6.40-alpha.312180.0<br/>@codecademy/gamut-labs@28.6.3-alpha.312180.0<br/>@codecademy/gamut@42.3.1-alpha.312180.0<br/>@codecademy/monaco@0.5.29-alpha.312180.0<br/>@codecademy/styleguide@52.5.2-alpha.312180.0<br/></pre>

**Comment by aresnik11** [comment]
@christian-dinh yeah that would be nice but unfortunately isnt super easy to do since we would only want to default `truncateLines` to 1 if `truncate` was passed. i'll make a follow up ticket for us or gamut team to look into it but am going to push forward for now

**Comment by christian-dinh** [comment]
yeah I assume getting types to cooperate would be tricky, nw!

**Pull request merged**